### ISSUE TITLE
Perf: ARTS `rotate` is ~100% of the 2D cost and discards 127/128 of its own output

### ISSUE THREAD

**Issue by joglekara** [ergodicio/tsadar#120]
## Summary

In the ARTS (2D EDF) path, `FormFactor.calc_chi_vals` calls `self.rotate` (`form_factor.py:428`), which runs `interpax.interp2d(..., method="cubic")` over all `nvx**2` grid points — for **every** evaluation point.

The very next line collapses it:

```python
fe_2D_k = checkpoint(self.rotate)(vx, DF, element * 180 / jnp.pi, reshape=False)
fe_1D_k = jnp.sum(fe_2D_k, axis=0) * dvx          # <- form_factor.py:429
```

Only the 1D **projection** is ever used. The full 2D rotation is thrown away. That projection is the Radon transform of the EDF at angle `beta`, and `beta` is the only per-point input to `rotate`.

## Evidence

Measured (JAX CPU, x64, `nvx=128`, via `jax.lax.map` with `batch_size=128` as in `_calc_all_chi_vals_`):

```
full calc_chi_vals       : 2266 us/point
  rotate only            : 2642 us/point   (~100%)
  ratintn only           :    4 us/point   (0.2%)
```

Everything other than `rotate` is noise.

`calc_in_2D` evaluates at `num_grad x npts x n_angles`. With the OMEGA ARTS geometry — `sa = np.arange(19, 139.5, 0.5)` -> 241 angles (`calibration.py:639`) — and a 1024-point wavelength axis, that is ~2.5e5 points, i.e. roughly **10 minutes of pure interpolation per forward pass on CPU**, plus the backward pass.

## Proposed fix: precompute the sinogram, interpolate in beta

Compute the projection at `NB` fixed angles once per forward pass, then look up each evaluation point in `(beta, v)`:

| | per-point cost |
|---|---|
| current | 2266 us |
| sinogram lookup | 0.33 us |
| **speedup** | **~6800x** |

Accuracy vs. the exact per-point rotation (linear interpolation in `beta`):

| `NB` | max rel err |
|---|---|
| 128 | 5.5e-04 |
| 256 | 1.4e-04 |
| 512 | 3.4e-05 |
| 1024 | 9.2e-06 |

`NB` becomes a tolerance knob. This also shrinks the AD graph from ~2.5e5 rotations to `NB` rotations, which should relieve the `checkpoint` pressure and the XLA compile cost that motivated #102.

Going further: the Fourier slice theorem builds the whole sinogram from a single FFT2 of the EDF plus `NB` radial slices, if the `NB` rotations themselves become the bottleneck.

## Secondary: uniform-grid indexing inside the interpolation

`interpax.interp2d` locates indices with `jnp.searchsorted`, but `vx` is a `linspace`. A uniform-grid bicubic using `floor((x-x0)/dx)` measured **3.4x** faster on CPU (2231 -> 653 us/point).

**Caveat, important:** ~20% of the rotated query points fall outside the velocity grid, so the result is genuinely sensitive to the extrapolation convention. A quick Catmull-Rom port of mine differed from `interpax` by **40%** on the projection for exactly this reason. Any replacement must reproduce `extrap=True` behavior faithfully and be tested against the current implementation, not just spot-checked in the interior.

## On CUDA / Pallas

A fused rotate+project+gradient Pallas kernel (holding the `nvx x nvx` EDF tile in SMEM) is the natural custom-kernel target here. But it should come **after** the sinogram reformulation — that change removes ~99.99% of the work, and what remains likely doesn't justify a hand-written kernel.

## Related

- #51 — parallelizing angular across GPUs. Complementary, but this change reduces the cost by ~3 orders of magnitude, which may make the multi-GPU path unnecessary for typical runs.
- #102 vectorized the ATS reductions to fix XLA compile blowup; this addresses the remaining cost inside the per-point kernel.

## Notes

Benchmarked on CPU (no GPU available in the session). Absolute timings will differ on GPU; the ~100%-of-cost attribution and the algorithmic redundancy are hardware-independent.
